apps/codeforafrica/src/lib/data/utils/posts.js

```diff
@@ -0,0 +1,185 @@
+import formatDate from "@/codeforafrica/utils/formatDate";
+
+export function formatTags(tags) {
+  const excludedTags = new Set(["stories", "opportunities"]);
+
+  const tagCounts = tags.reduce((counts, { name }) => {
+    // eslint-disable-next-line no-param-reassign
+    counts[name] = (counts[name] || 0) + 1;
+    return counts;
+  }, {});
+
+  const sortedTags = Object.keys(tagCounts)
+    .filter((tag) => !excludedTags.has(tag.toLowerCase()))
+    .sort((a, b) => tagCounts[b] - tagCounts[a])
+    .map((tagName) => {
+      const tag = tags.find(({ name }) => name === tagName);
+      return {
+        name: tag.name,
+        slug: tag.slug,
+      };
+    });
+
+  return sortedTags;
+}
+
+export function formatPost(post, primaryTag) {
+  const {
+    id,
+    title,
+    coverImage: { src, alt },
+    excerpt,
+    slug,
+    publishedOn,
+    tags,
+  } = post;
+  if (!title) {
+    return null;
+  }
+  return {
+    id,
+    title,
+    image: { src, alt },
+    excerpt,
+    publishedOn: formatDate(publishedOn, {
+      includeTime: false,
+      month: "short",
+    }),
+    tags: formatTags(tags),
```

Comment:
Do we format tags for each individual post?

Reply:
@kilemensi Yes. We need to remove the excused tags from the list

Reply:
Removing primary tag is different from formatting. Removing primary tag could be a simple as skipping it in the front-end.

Formatting function sorts, etc., and if there are many posts with a lot of tags, this could slow. My initial understanding was may be only the main tags filter on the stories would need this sorting
<img width="779" alt="Screenshot 2023-09-28 at 16 12 38" src="https://github.com/CodeForAfrica/ui/assets/1779590/9a0192b0-19a1-4d6c-841a-43554734841f">

... thinking about it more, not sure whether sorting these tags by count vs alphabetically would make more sense to site visitors.

Reply:
Sorting alphabetically should be the best way